components/WeightComponent/RealWeight.cpp

```diff
@@ -1,28 +1,33 @@
-#include "RealWeight.hpp"
+#include "include/RealWeight.hpp"
 
 using namespace Weight;
 
 RealWeight::RealWeight()
 {
-    hx711_t dev = {
-        .dout = GPIO_NUM_19,
-        .pd_sck = GPIO_NUM_18,
-        .gain = HX711_GAIN_A_128
+    hx711_t devs[4] = {
+        { .dout = GPIO_NUM_19, .pd_sck = GPIO_NUM_18, .gain = HX711_GAIN_A_128 },
+        { .dout = GPIO_NUM_21, .pd_sck = GPIO_NUM_22, .gain = HX711_GAIN_A_128 },
+        { .dout = GPIO_NUM_23, .pd_sck = GPIO_NUM_25, .gain = HX711_GAIN_A_128 },
+        { .dout = GPIO_NUM_26, .pd_sck = GPIO_NUM_27, .gain = HX711_GAIN_A_128 }
     };
 
-    mRealSensor = dev;
-
-    hx711_init(&mRealSensor);
+    for (int i = 0; i < 4; ++i) {
+        mRealSensor[i] = devs[i];
+        hx711_init(&mRealSensor[i]);
+    }
 }
 
 int32_t RealWeight::getWeight()
 {
-    int32_t data = 0;
-    esp_err_t r = hx711_read_average(&mRealSensor, 10, &data);
-    //esp_err_t r = hx711_read_data(&mRealSensor, &data);
-    if (r == ESP_OK) {
-        return data;
+    int32_t data[4] = {0};
+    int32_t totalWeight = 0;
+
+    for (int i = 0; i < 4; ++i) {
+        esp_err_t r = hx711_read_average(&mRealSensor[i], 10, &data[i]);
```

Comment:
on a similar note as earlier, the read average function is also for only one hx711 module (two pins and one hx711_t struct). this is beacuse the read average function simply reads several values across time and averages those, not several values from multiple sensors. if you're getting garbacge values / timeouts / device not connected-type messages this is probably why, since some of those pins are being used for other devices / are not connected to anything / if you're on the esp32-cam are connected internally.

Reply:
i think the previous code was actually correcte (read average, write that into data, then return data only if there is no error)